replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/MetacardConfigLoader.java

```diff
@@ -54,97 +54,80 @@
 
   private final MetacardHelper helper;
 
-  private final MetacardType configMetacardType;
-
   private final SiteManager siteManager;
 
+  private final ReplicatorConfigManager newConfigManager;
+
+  private final Security security;
+
   public MetacardConfigLoader(
       CatalogFramework framework,
       FilterBuilder filterBuilder,
       MetacardHelper helper,
-      MetacardType configMetacardType,
-      SiteManager siteManager) {
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager) {
+    this(framework, filterBuilder, helper, siteManager, newConfigManager, Security.getInstance());
+  }
+
+  @VisibleForTesting
+  MetacardConfigLoader(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      Security security) {
     this.framework = framework;
     this.filterBuilder = filterBuilder;
     this.helper = helper;
-    this.configMetacardType = configMetacardType;
     this.siteManager = siteManager;
+    this.newConfigManager = newConfigManager;
+    this.security = security;
   }
 
-  public ReplicatorConfig createConfig() {
-    return new ReplicatorConfigImpl();
+  public void init() {
+    Failsafe.with(
+            new RetryPolicy()
+                .withBackoff(3, 60, TimeUnit.SECONDS)
+                .withMaxDuration(10, TimeUnit.MINUTES))
```

Comment:
❓ Why are we stopping at 10 minutes?  Shouldn't we be continuing until it works?
❓ Should we not be re-trying for errors other than when Solr is not available?

Reply:
If solr doesn't come up 10 minutes after this bundle does than I didn't think It'd be likely to come up at all but I guess I don't see a good reason for not just continuing to retry every minute. To answer your 3rd question I think this retries for any error that occurs, not just solr.

Reply:
On the contrary, the system is designed to continuously retry to connect to Solr in the background without having the admin come in and be forced to restart us. Remember that we support not just the Solr instance we start from the `ddf` script but we also support Solr cloud or an admin managed Solr instance. You should therefore continue retrying. That is if the error is Solr not being available versus another error for which retry will never ever help.